pkg/tools/etcd_tools.go

```diff
@@ -418,20 +418,24 @@ func (h *EtcdHelper) AtomicUpdate(key string, ptrToType runtime.Object, ignoreNo
 	}
 }
 
-func checkEtcd(host string) error {
+// Performs a version check against the provided Etcd server, returning a triplet
+// of the release version, internal version, and error (if any).
+func CheckEtcd(host string) (string, string, error) {
```

Comment:
I'll get a new PR with the named return params and change the method name. As far as who's going to consume these, I have a separate change to kube2sky that uses it as a check that etcd is running. Note, though, that none of the code does a minimum version check (yet). I figure this may wait until it's needed.